server/api/dataset.py

```diff
@@ -425,19 +425,14 @@ def _checkFileFormat(self, file, formats):
     )
     @access.cookie
     @access.public(scope=TokenScope.DATA_READ)
-    @loadmodel(model='file', map={'metadataFileId': 'metadataFile'}, force=True, exc=False)
+    @loadmodel(model='file', map={'metadataFileId': 'metadataFile'}, level=AccessType.READ,
+               exc=False)
```

Comment:
@brianhelba This line's change requires https://github.com/girder/girder/pull/2716. Although I think it's good defensive programming, I'm open to backing this change out since the access checks performed here are redundant: assuming metadata files will always attached to datasets, then the below access check on the dataset is sufficient.